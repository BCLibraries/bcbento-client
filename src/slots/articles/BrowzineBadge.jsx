import React from 'react';

/**
 * Links to FullText in Browzine
 *
 * @param {object} libkeyAvailability
 * @param {string} libkeyAvailability.fullTextFile
 * @param {string} libkeyAvailability.browzineWebLink
 * @constructor
 */
function BrowzineBadge({libkeyAvailability}) {
    return (
        <div className="browzine-badge">
            {libkeyAvailability.fullTextFile &&
            <a href={libkeyAvailability.fullTextFile} className="browzine-badge__download-now" target="_blank" rel="noopener noreferrer">
                <img src="https://assets.thirdiron.com/images/integrations/browzine-pdf-download-icon.svg" className="browzine-pdf-icon" aria-hidden="true" width="12" height="16" alt=""/>
                <span className="browzine-web-link-text">Download Now
                    <span className="browzine-badge__pdf_notice"> (PDF)</span>
                </span>
            </a>
            }

            {libkeyAvailability.browzineWebLink &&
            <a className="browzine-badge__view-issue" href={libkeyAvailability.browzineWebLink}  target="_blank" rel="noopener noreferrer">
                <img src="https://assets.thirdiron.com/images/integrations/browzine-open-book-icon.svg" className="browzine-book-icon" aria-hidden="true" width="15" height="15" alt=""/>
                <span className="browzine-web-link-text">View Issue Contents</span>
            </a>
            }
        </div>
    );
}

export default BrowzineBadge;
