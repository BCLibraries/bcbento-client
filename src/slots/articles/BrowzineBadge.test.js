import renderer from 'react-test-renderer';
import BrowzineBadge from './BrowzineBadge.jsx';

it('renders correctly', () => {
    const libkeyAvailability = {
        "fullTextFile": "https://libkey.io/libraries/431/articles/4749574/full-text-file?utm_source=api_536",
        "contentLocation": "https://libkey.io/libraries/431/articles/4749574/content-location?utm_source=api_536",
        "availableThroughBrowzine": true,
        "browzineWebLink": null,
        "journals": [
            {
                "browzineEnabled": true,
                "browzineWebLink": "https://browzine.com/libraries/431/journals/8166?utm_source=api_536",
            }
        ],
    }
    const component = renderer.create(
        <BrowzineBadge libkeyAvailability={libkeyAvailability} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
