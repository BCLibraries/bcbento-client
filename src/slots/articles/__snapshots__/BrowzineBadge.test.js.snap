// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="browzine-badge"
>
  <a
    className="browzine-badge__download-now"
    href="https://libkey.io/libraries/431/articles/4749574/full-text-file?utm_source=api_536"
    rel="noopener noreferrer"
    target="_blank"
  >
    <img
      alt=""
      aria-hidden="true"
      className="browzine-pdf-icon"
      height="16"
      src="https://assets.thirdiron.com/images/integrations/browzine-pdf-download-icon.svg"
      width="12"
    />
    <span
      className="browzine-web-link-text"
    >
      Download Now
      <span
        className="browzine-badge__pdf_notice"
      >
         (PDF)
      </span>
    </span>
  </a>
</div>
`;
